### PR TITLE
Add a sample executable that hogs up all the CPU

### PR DESCRIPTION
**Pull request by emersonknapp**
* Add a new `ament_cmake` ROS2 package `example_nodes`
* Create a process that hogs a configurable amount of CPU by forking busy processes
* Copy the new package into the Dockerfile so it is built
* Reorder the Dockerfile a little so it is more friendly to rebuild when code is changed
* Add instructions in README for a demo that makes everything slow to a crawl (and fix up some existing issues in the sandbox instructions)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -157,19 +157,33 @@ Build the docker image required using the dockerfile from this repository
 ``` bash
 git clone git@github.com:ros-swg/turtlebot3_demo.git
 cd turtlebot3_demo
-docker build . -t rosswg:turtlebot3_sandbox_demo
+docker build . -t ros-swg/turtlebot3_demo
```
Comment (ruffsl) [inline README.md]: The docker repo for this github repo is `rosswg/turtlebot3_demo` , so we should probably be consistent on the latest tag we use to reference this Dockerfile, whether users build it themselves, or pull it from our Docker repo to skip the build time.

Comment (emersonknapp) [reply README.md]: Good catch! I was trying to change the existing readme to match the upstream tag - I messed it up, will change

File: example_nodes/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+find_package(ament_cmake REQUIRED)
+find_package(rcutils REQUIRED)
+
+add_executable(cpu_hog src/cpu_hog.cpp)
+target_include_directories(cpu_hog PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+ament_target_dependencies(
+  cpu_hog
+  "rcutils"
```
Comment (piraka9011) [inline example_nodes/CMakeLists.txt:28]: Does this need to be a string? Just looking for consistency?

**Review by piraka9011 — APPROVED** [review]

File: example_nodes/CMakeLists.txt (hunk 2 above)
Comment (emersonknapp) [reply example_nodes/CMakeLists.txt:28]: args to ament_target_dependencies are in quotes like this in every ament_cmake package i've seen - i couldn't tell you exactly why

File: README.md [hunk 3]
```diff
@@ -141,33 +141,46 @@ export ROS_SECURITY_ENABLE=true
...
-## Running the Security Workshop Sandbox Node Demo
+### Running the Security Workshop Sandbox Node Demo
+
+``` bash
+rocker --x11 --nvidia rosswg/turtlebot3_demo "byobu -f configs/sandbox_demo/unsecure.conf attach"
+```
+
+Omit the `--nvidia` arg if you don't have dedicated GPU for hardware acceleration of 3D OpenGL views.
 
-Build the docker image required using the dockerfile from this repository
```
Comment (dabonnie) [inline README.md:153]: Is it clear that the docker image will need to be built? I don't think we're hosting the new image, so doesn't it have to be generated locally?

File: README.md [hunk 4]
```diff
@@ -141,33 +141,46 @@ export ROS_SECURITY_ENABLE=true
...
-Omit the `--nvidia` arg if you don't have dedicated GPU for hardware acceleration of 3D OpenGL views.
+After launching with the above config, there is a command ready in the `bad_actor` byobu window, ready to run a cpu hog that will fork many busy processes (`ros2 run example_nodes cpu_hog 8`).
+This is a contrived example - but it illustrates what could happen if a node that you are using hits an untested code path and goes into an infinite loop.
+If you run the command, you can see the CPU of your system jump to 100% usage, grinding everything else to a halt.
 
-Likewise, the following command is used to run the demo using the secure config:
+UPCOMING: Show usage of `launch_ros_sandbox` resource limits to keep the demo running happily.
+
+## Developing
+To rebuild this demo locally if you are working on it, you can rebuild the Docker image with the same tag, so all above demo commands will work correctly.
```
Comment (dabonnie) [inline README.md:180]: Per the comment above, this seems like a requirement to run the demo rather than solely developing.

File: README.md (hunk 3 above)
Comment (emersonknapp) [reply README.md:153]: No - if you look at the instruction there is absolutely no difference between what you would create locally and what we would be building upstream. The instructions provide no special flags, no separate Dockerfile. It's exactly the same image that we're hosting. We should make sure the Dockerhub repo is set up to autobuild the image from master here - it's really easy to do.

Comment (dabonnie) [reply README.md:153]: Does the hosted docker image contain the launch-ros-sandbox dependency? Notice the https://github.com/ros-swg/turtlebot3_demo/blob/master/.docker/overlay.repos: it now contains that dependency (my last PR: https://github.com/ros-swg/turtlebot3_demo/pull/14).

Comment (emersonknapp) [reply README.md:153]: @ruffsl is the https://hub.docker.com/r/rosswg/turtlebot3_demo/tags image set to auto-build `latest` from `master` here? If not, can we do that?

Comment (emersonknapp) [reply README.md:153]: @dabonnie it will once we update the image on hub

Comment (ruffsl) [reply README.md:153]: Let's keep the `roscon19` tag as an image we manually curate and check for the tutorial. However, I'll enabled auto build for the master branch to build to the implicit `latest` tag, so pulling `rosswg/turtlebot3_demo` should pull an image derived from the latest commit.

I've having some docker cloud linking issues, so I'll resolve that later today.

Comment (emersonknapp) [reply README.md:153]: @ruffsl any luck with linking dockerhub? happy to help if needed

File: configs/sandbox_demo/bad_actor.conf [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+# setup secure session
+new-session -s secure -n diagnostic -d
+send-keys 'glances' Enter
+
+setenv FOO "foo"
+
+source configs/sandbox_demo/security_workshop_demo_common.conf
+
+new-window -n bad_actor
+send-keys 'ros2 run example_nodes cpu_hog 8'
```
Comment (dabonnie) [inline configs/sandbox_demo/bad_actor.conf]: nit: newline

**Review by dabonnie — APPROVED** [review]
LGTM

**Pull request merged**